armi/physics/neutronics/crossSectionSettings.py

```diff
@@ -125,65 +181,136 @@ def isPregenerated(self):
         """True if this points to a pre-generated XS file."""
         return self.fileLocation is not None
 
+    def _validate(self):
+        """
+        Perform validation checks on the set of inputs.
+
+        This makes some design assumptions about the lattice
+        physics code (i.e., critical buckling search is not
+        a valid option for non-0D problems), so if different
+        behavior is desired then it is recommended to subclass
+        this and to make a new cross section setting.
+
+        Notes
+        -----
+        This checks for any inconsistencies in the definition of
+        the inputs.
+        """
+        if self.fileLocation is None and self.geometry is None:
+            raise ValueError(f"{self} is missing a geometry input or a file location.")
+
+        if self.fileLocation is not None and self.geometry is not None:
+            raise ValueError(
+                f"The file location and geometry inputs cannot be combined in {self}. "
```

Comment:
slight clarification to explain what to do could be like: "Either file location or geometry inputs must be given, but not both. Remove one or the other"

Reply:
Updated in revision.